atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbRuntimeConfig.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import javax.annotation.Nullable;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonDeserialize(as = ImmutableAtlasDbRuntimeConfig.class)
+@JsonSerialize(as = ImmutableAtlasDbRuntimeConfig.class)
+@Value.Immutable
+public abstract class AtlasDbRuntimeConfig {
+    /**
+     * If true, a background thread will periodically delete cells that
+     * have been overwritten or deleted. This differs from scrubbing
+     * because it is an untargeted cleaning process that scans all data
+     * looking for cells to delete.
+     */
+    // TODO(ssouza): make this Default after removing support for mirror method on the AtlasDbConfig.
+    @Nullable
+    public abstract Boolean enableSweep();
+
+    /**
+     * The number of milliseconds to wait between each batch of cells
+     * processed by the background sweeper.
+     */
+    // TODO(ssouza): make this Default after removing support for mirror method on the AtlasDbConfig. We need to detect
+    // the absence of this value for now.
+    @Nullable
+    public abstract Long getSweepPauseMillis();
```

Comment:
drive-by: From my internal projects I can suggest that this kind of config becomes sub-optimal as soon as something else becomes live reloadable - e.g. it's better to have

```yaml
atlas-runtime:
    sweep:
        enabled: true
        readLimit: 1234
        candidateBatchHint: 50
        deleteBatchHint: 100
    timestamp:
        batchTarget: 100
        requestLimitPerSecond: 200
```

rather than

```yaml
atlas-runtime:
    sweepEnabled: true
    sweepReadLimit: 1234
    sweepCandidateBatchHint: 50
    sweepDeleteBatchHint: 100
    timestampBatchTarget: 100
    timestampRequestLimitPerSecond: 200
```

- definitely don't view either of my comments as anything but random comments, just my first impression while reading it.